plugins/default/check_mongodb_wrapper.sh

```diff
@@ -0,0 +1,110 @@
+#!/bin/bash
+######################################################################
+#
+# Nagios wrapper plugin to retrieve mongo admin password from
+# same project where nagios is located and then perform the
+# check requested
+#
+######################################################################
+
+################################
+# GLOBALS
+################################
+# Nagios exit codes
+NAGIOS_OK=0
+NAGIOS_WARN=1
+NAGIOS_CRIT=2
+NAGIOS_UNKNOWN=3
+
+################################
+# FUNCTIONS
+################################
+usage() {
+ echo "Usage:"
+ echo " -w <warn>    Warning threshold (default 0)"
+ echo " -c <crit>    Critical threshold (default 0)"
+ echo " -A <action>  Action as defined"
+ echo " -S <name>     OpenShift selector name"
+ echo " -h           Show this message"
+ echo "Example: $0 -A connect -S mongodb-server -w 0 -c 0"
+}
+
+################################
+# ARGUMENT PROCESSING
+################################
+WARN=0
+CRIT=0
+ACTION=
+SELECTOR=mongodb-replica
+while getopts "h:w:c:A:S:" OPTION
+do
+ case $OPTION in
+  h) usage; exit $NAGIOS_UNKNOWN ;;
+  w) WARN=$OPTARG ;;
+  c) CRIT=$OPTARG ;;
+  A) ACTION=$OPTARG ;;
+  S) SELECTOR=$OPTARG ;;
+  ?) usage; exit $NAGIOS_UNKNOWN ;;
+ esac
+done
+
+# Ensure all expected variables are populated
+if [[ -z $ACTION ]] || [[ -z $SELECTOR ]] || [[ -z $WARN ]] || [[ -z $CRIT ]]; then
+ usage; exit $NAGIOS_UNKNOWN
+fi
+
+# Use a temp file to store list of pods with their mongodb connection info
+TEMP_FILE=`mktemp`
+oc get pod --selector=name=$SELECTOR --output json | jq '.items[] | select(.status.phase == "Running") | [.metadata.name, .status.podIP, .spec.containers[].ports[].containerPort, ( .spec.containers[].env[] | select(.name == "MONGODB_ADMIN_PASSWORD") | .value) ] | @tsv' -r > $TEMP_FILE
+
+# The array to store the results for each pod being checked
+declare -a RESULT_ARR=()
+
+# Other prep variables
+COUNT_WARN=0
+COUNT_CRIT=0
+COUNT_OK=0
+CURR=1
+
+# Iterate through each mongodb member, run the check and collect the results 
+while IFS='' read -r line || [[ -n "$line" ]]; do
+  POD=`echo $line | awk '{print $1}'`
+  IP=`echo $line | awk '{print $2}'`
+  PORT=`echo $line | awk '{print $3}'`
+  MONGO_PASS=`echo $line | awk '{print $4}'`
+
+  RESULT_ARR[$CURR]="$POD - `/opt/rhmap/nagios/plugins/check_mongodb.py -H $IP -P $PORT -C $CRIT -W $WARN -A $ACTION -u admin -p $MONGO_PASS 2>&1`"
+
+  CURRENT_RESULT=$?
```

Comment:
This script doesn't set the common options*, and makes use of `$?` to get the exit code of `check_mongodb.py` (I hope that is, and will continue to be, the last command :) ).

I find it to be error prone, specially alarming for something that will be our nagios check.

For instance, there's a `oc get pod` call above, that may fail and the script will continue running.